pkg/rbac/sharedData.go

```diff
@@ -51,16 +55,86 @@ var managedClusterResourceGvr = schema.GroupVersionResource{
 	Resource: "managedclusters",
 }
 
-func (cache *Cache) PopulateSharedCache(ctx context.Context) error {
+func (shared *SharedData) GetPropertyTypes() (map[string]string, error) {
+
+	// original query:
+	// select distinct key, jsonb_typeof(value) as datatype
+	// FROM search.resources,jsonb_each(data)
+	// order by datatype;
+
+	var selectDs *goqu.SelectDataset
+	//define schema:
+	schemaTable := goqu.S("search").Table("resources")
+	//data expression to get value and key
+	jsb := goqu.L("jsonb_each(?)", goqu.C("data"))
+	//select from these datasets
+	ds := goqu.From(schemaTable, jsb)
+	//select statement with orderby and distinct clause
+	selectDs = ds.Select(goqu.L("key"), goqu.L("jsonb_typeof(?)",
+		goqu.C("value")).As("datatype")).Distinct().Order(goqu.L("datatype").Asc())
+
+	query, params, err := selectDs.ToSQL()
+
+	klog.V(5).Info("Query for property datatypes: [%s] ", query)
+	rows, err := CacheInst.pool.Query(context.Background(), query, params...)
```

Comment:
This function should get a context as an argument instead of creating a new one here.